GAE/src/test/java/org/akvo/flow/util/FlowJsonObjectReaderTests.java

```diff
@@ -0,0 +1,180 @@
+                                        "      \"surveyList\": null\n" +
+                                        "    }\n" +
+                                        "  ], \n" +
+                                        "  \"message\": null, \n" +
+                                        "  \"offset\": 0, \n" +
+                                        "  \"resultCount\": 0, \n" +
+                                        "  \"url\": null\n" +
+                                        "}\n";
+
+    @Test
+    void testReadSimpleJsonObject() {
+        FlowJsonObjectReader reader = new FlowJsonObjectReader();
+        TypeReference<QuestionDto> typeReference = new TypeReference<QuestionDto>() {};
+        QuestionDto testQuestionDto = null;
+        try {
+            testQuestionDto = reader.readObject(QUESTION_DTO_JSON_STRING, typeReference);
+        } catch (IOException e) {
+            //
+        }
+        assertEquals(testQuestionDto.getType(), QuestionDto.QuestionType.FREE_TEXT);
+        assertEquals(testQuestionDto.getText(),"How many toilets are present?");
+        assertFalse(testQuestionDto.getDependentFlag());
+        assertEquals(testQuestionDto.getQuestionGroupId(), 12345678L);
+        assertEquals(testQuestionDto.getType(), QuestionDto.QuestionType.FREE_TEXT);
+        assertEquals(testQuestionDto.getOrder(), 0);
+    }
+
+    @Test
+    void testReadComplexJsonObject() {
+        FlowJsonObjectReader reader = new FlowJsonObjectReader();
+        TypeReference<Map<String, List<CaddisflyResource>>> typeReference = new TypeReference<Map<String, List<CaddisflyResource>>>() {};
+        Map<String, List<CaddisflyResource>> resourcesMap = new HashMap<>();
+        try {
+            resourcesMap = reader.readObject(COMPLEX_JSON_OBJECT, typeReference);
+        } catch (IOException e) {
+            //
+        }
+
+        List<CaddisflyResource> resourcesList = resourcesMap.get("tests");
+        assertNotEquals(resourcesList, null);
+        assertEquals(resourcesList.size(), 2);
+        assertEquals(resourcesList.get(0).getResults().size(), 2);
+        assertEquals(resourcesList.get(0).getName(), "Soil - Electrical Conductivity");
+        assertEquals(resourcesList.get(1).getResults().size(), 1);
+        assertEquals(resourcesList.get(1).getUuid(), "0b4a0aaa-f556-4c11-a539-c4626582cca6");
+    }
+
+    @Test
+    void testDtoListResponses() {
+        FlowJsonObjectReader reader = new FlowJsonObjectReader();
+        TypeReference<SurveyGroupDto> typeReference = new TypeReference<SurveyGroupDto>() {};
+        List<SurveyGroupDto> surveyList = null;
+
+        try {
+            surveyList = reader.readDtoListObject(DTO_LIST_JSON_OBJECT, typeReference);
+        } catch (IOException e) {
+            //
+        }
+
+        assertNotEquals(surveyList, null);
+        assertEquals(surveyList.size(), 1);
+        assertEquals(surveyList.get(0).getName(),"1.10.36 all questions");
```

Comment:
This is a minor detail but the the assertZZZ methods work like this assertZZZ(expected, actual), this is valid for the whole of your test files. Also please format the file as some "," have no space after them.

Reply:
https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html

Reply:
🤦‍♂️ no thats a major detail! 😛